sklearn/linear_model/_ransac.py

```diff
@@ -404,12 +428,33 @@ def fit(self, X, y, sample_weight=None):
         estimator_name = type(estimator).__name__
         if sample_weight is not None and not estimator_fit_has_sample_weight:
             raise ValueError(
-                "%s does not support sample_weight. Samples"
+                "%s does not support sample_weight. Sample"
                 " weights are only used for the calibration"
                 " itself." % estimator_name
             )
+
         if sample_weight is not None:
-            sample_weight = _check_sample_weight(sample_weight, X)
+            fit_params["sample_weight"] = sample_weight
+
+        if _routing_enabled():
+            try:
+                routed_params = process_routing(self, "fit", **fit_params)
+            except UnsetMetadataPassedError as e:
+                raise UnsetMetadataPassedError(
+                    message=(
+                        f"{e}, which is used internally by `RANSACRegressor.fit()`. "
+                        f"Call `{estimator.__class__.__name__}.set_{{method}}_request("
+                        "{metadata}=True)` for each metadata."
+                    ),
+                    unrequested_params=e.unrequested_params,
+                    routed_params=e.routed_params,
+                ) from e
```

Comment:
can't find the previous messages about this.

I don't think we should modify the exception here. The exception, if needs to be improved, needs to be handled for every estimator, and we shouldn't expect this much boilerplate from meta-estimators.